pypeit/coadd3d.py

```diff
@@ -0,0 +1,1772 @@
+                   specname="PYP_SPEC"):
+        """
+        Create a WCS and the expected edges of the voxels, based on user-specified
+        parameters or the extremities of the data. This is a convenience function
+        that calls the core function in `pypeit.core.datacube`_.
+
+        Parameters
+        ----------
+        all_ra : `numpy.ndarray`_
+            1D flattened array containing the RA values of each pixel from all
+            spec2d files
+        all_dec : `numpy.ndarray`_
+            1D flattened array containing the DEC values of each pixel from all
+            spec2d files
+        all_wave : `numpy.ndarray`_
+            1D flattened array containing the wavelength values of each pixel from
+            all spec2d files
+        dspat : float
+            Spatial size of each square voxel (in arcsec). The default is to use the
+            values in cubepar.
+        dwave : float
+            Linear wavelength step of each voxel (in Angstroms)
+        collapse : bool, optional
+            If True, the spectral dimension will be collapsed to a single channel
+            (primarily for white light images)
+        equinox : float, optional
+            Equinox of the WCS
+        specname : str, optional
+            Name of the spectrograph
+
+        Returns
+        -------
+        cubewcs : `astropy.wcs.WCS`_
+            astropy WCS to be used for the combined cube
+        voxedges : tuple
+            A three element tuple containing the bin edges in the x, y (spatial) and
+            z (wavelength) dimensions
+        reference_image : `numpy.ndarray`_
+            The reference image to be used for the cross-correlation. Can be None.
+        """
+        # Setup the cube ranges
+        reference_image = None  # The default behaviour is that the reference image is not used
+        ra_min = self.cubepar['ra_min'] if self.cubepar['ra_min'] is not None else np.min(all_ra)
+        ra_max = self.cubepar['ra_max'] if self.cubepar['ra_max'] is not None else np.max(all_ra)
+        dec_min = self.cubepar['dec_min'] if self.cubepar['dec_min'] is not None else np.min(all_dec)
+        dec_max = self.cubepar['dec_max'] if self.cubepar['dec_max'] is not None else np.max(all_dec)
+        wav_min = self.cubepar['wave_min'] if self.cubepar['wave_min'] is not None else np.min(all_wave)
+        wav_max = self.cubepar['wave_max'] if self.cubepar['wave_max'] is not None else np.max(all_wave)
+        if self.cubepar['wave_delta'] is not None:
+            dwave = self.cubepar['wave_delta']
+
+        return datacube.create_wcs(all_ra, all_dec, all_wave, dspat, dwave, ra_min=ra_min, ra_max=ra_max,
+                                   dec_min=dec_min, dec_max=dec_max, wave_min=wav_min, wave_max=wav_max,
+                                   reference=self.cubepar['reference_image'], collapse=collapse, equinox=equinox,
+                                   specname=specname)
+
```

Comment:
It looks like someone has to pass in the standard cube here rather than an archived sensitivity function. I'm not sure I follow the logic there. In PypeIt sensitivity function generation is always a separate step, and sensitivity functions are passed as arguments to functions, not as filenames in parsets. This thus constitutes a significant departure from the model we use for fluxing. I also don't see why you would want to re-extract the standard and recompute sensitivity functions every time you want to coadd in 3d. Perhaps I'm missing some of the intricacies of the complex control flow here, but please elaborate. To be clear, the question I'm asking is why not do this once and pass in the sensitivity function. Also, you are using the UV sensitivity function algorithm here, but that is about to be deprecated. So you should not be writing new code that depends on code that will be deprecated. We are transitioning to IR sensitivity function everywhere.

Reply:
I'm fine with leaving this as for now @rcooke-ast but can you at least make an issue for these changes. It does not make sense to me to have sensitivity function computation as part of a coadd algorithm.